docs/source/features/parallelisms.rst

```diff
@@ -80,10 +72,36 @@ The distributed optimizer in NeMo is built on top of
 `DistributedFusedAdam <https://github.com/NVIDIA/apex/blob/master/apex/contrib/optimizers/distributed_fused_adam.py>`_
 from Apex.
 
+Fully-Shared Data Parallelism (FSDP)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+NeMo supports Fully-Sharded Data Parallelism (FSDP) that shards parameter gradients and low-precision parameters for computation on top of the model states that Distributed optimizer shards (optimizer states and high-precision parameters).
+Since FSDP shards the entire model states, it ensures linear model state memory saving with increasing DP size.
+FSDP can be preferred for the LLM training with unbalanced workload between pipeline stages (or Transformer layers) or with a large vocabulary size, where pipelining would cause huge computation bubbles due to the workload imbalance.
+Also, FSDP unloads the effort to search the performance-optimal mappings with 3D parallelism (TP/PP/DP) because it has a single parallelization domain.
+
+NeMo uses `pytorch's FSDP interface <https://pytorch.org/tutorials/intermediate/FSDP_tutorial.html>`_ to shard LLM model states, which flattens the parameters of each Transformer layer and partitions across datap-parallel GPUs.
+FSDP introduces collectives across data-parallel GPUs; all-gather of the parameters for computation and reduce-scatter of parameter gradients.
+The parameter all-gather happens in both network forward- and back-propagation phases, and the gradient reduce-scatter happens only in the back-propagation.
+These FSDP communications are overlapped with Transformer layer computations.
+
+Setting ``fsdp=true`` enables FSDP.
+The mixed precision recipe can be set by ``precision`` knob, which determines both the computation and communication precisions.
+Also, one can use ``grad_reduce_dtype`` to override the gradient reduction precision specifically.
+
+
+Model Parallelism
+-----------------
+
+Model parallelism (MP) is a distributed model deployment method that partitions the model parameters across GPUs to reduce the need of per-GPU memory.
+NeMo supports various model-parallel methods, which can be mixed to maximize LLM training performance.
+
 Tensor Parallelism
 ^^^^^^^^^^^^^^^^^^
 
-Tensor Parallelism (TP) is a method for distributing a model's computation across multiple GPUs by splitting tensors into non-overlapping pieces. This allows different parts of the tensor to be processed simultaneously on separate GPUs, enhancing performance and enabling the training of larger models.
+Tensor Parallelism (TP) is a model-parallel partitioning method that distributes the parameter tensor of an individual layer across GPUs.
+On top of reducing the model state memory usage, it also saves the activation memory as per-GPU tensor sizes shrinks.
+However, the reduced per-GPU tensor lowers per-GPU-kernel workload sizes that increases CPU overhead.
```

Comment:
Edits to Line 14

To enable TP in the NeMo Framework, configure the ``tensor_model_parallel_size`` parameter in the model configuration. This parameter determines the number of GPUs among which the model's tensors are partitioned.